README.md

```diff
@@ -467,7 +468,70 @@ function App() {
 }
 ```
 
-### Theming
+## sx prop
+
+A special `sx` prop lets you apply styles directly to an element. When `sx` prop is specified on a component, Pigment CSS will replace any `sx` prop with `className` and `style` props at build time.
+
+The `sx` prop works on any component, including HTML elements and 3rd-party custom components as long as it is JSX.
+
+### Usage
+
+```js
+<div sx={{ display: 'flex', flexDirection: 'column' }}>
+```
+
+For a React component, it must spread the `className` and `style` to the underlying DOM element, otherwise the styles won't be applied.
+
+```js
+const AnyComponent = (props) => {
+  return <div {...props} />;
+};
+
+<AnyComponent sx={{ fontSize: 12, color: 'red' }} />;
+```
+
+> 💡 **Gotcha**: You don't need to pass `sx` prop to underlying component or DOM because it will be replaced with `className` and `style` props.
+
+The value provided to `sx` prop can be one of the following:
+
+- a plain style object (recommended)
+- a callback function that receives the [theme object](#theming) then return a plain style object:
+
+  ```js
+  <div sx={(theme) => ({ color: theme.colors.primary })} />
```

Comment:
Maybe we can add example that is not possible with an object, so that people are aware of what's the different use-case.

Reply:
What do you mean by "not possible with an object"? Can you give me an example?